TradeShare490/TradeShare
# [TS-51] Follow List and Follow/Unfollow Action

Video: https://youtu.be/tCwbBwt8U6I
There are a lot of console.log and 1 button [State] inside otherdashboard, they are removed.

TS-51
![image](https://user-images.githubusercontent.com/37958655/158036518-9df7f3b1-a23d-482c-96d8-106a39b3cb8a.png)

+ Login state for follower/following
+ Middleware for follow related action
+ Live following/follower
+ Connect other dashboard with backend (Follow, Position)
+ Update btn style at linking 3rd party APP


New dependency: vuex-persistedstate

OLD:
This PR contains TS-327 and TS-397, move TS-328 to next sprint
![image](https://user-images.githubusercontent.com/37958655/154883372-5718d9d9-6f60-4e4b-981f-bb7e786e11b1.png)

Followings
![image](https://user-images.githubusercontent.com/37958655/154880197-fca63377-eb19-4b43-bf1d-766edd941bcb.png)
Followers
![image](https://user-images.githubusercontent.com/37958655/154880208-52b97628-0959-417a-9909-14e3dbdf5850.png)

Snackbar notification for the success response
![image](https://user-images.githubusercontent.com/37958655/154880261-d5888964-b185-4b0c-b6bb-003d422198fb.png)

When list is empty
![image](https://user-images.githubusercontent.com/37958655/154880387-0a1a03a8-d16f-4f18-8c65-4dbfdf55b0ed.png)

Sidebar (fetch at login, dont know how to refresh, see Issues below)
![image](https://user-images.githubusercontent.com/37958655/154883445-2e4b7f74-7f0e-4179-8383-9ccb4f9b0488.png)

New asset
![image](https://user-images.githubusercontent.com/37958655/154880287-bc7738e4-02c7-4d0b-a8da-61d9822e8c0d.png)

Issue (I think we need a new ticket for these): 
- No profile picture implement, everything will be set to default (above asset)
- Slow when fetching the list
- Not sure how to refresh number in sidebar once follow/unfollow happened
- Remove dummy data, the tests are failing, I commented them out

Current relationship (API https://documenter.getpostman.com/view/10545271/UVe9QU3y#b55383f2-6f80-4271-be4f-a0514a28f0d6)
![image](https://user-images.githubusercontent.com/37958655/154880473-0a50b436-16b0-439a-bc0e-9b8752d113d2.png)

Accounts I use to test
test                 [6204a777f82a1fb5c9f80a1f]
TianMing        [6211cb412ab77bf00f22fd87]
TianMing2      [6211cb5a2ab77bf00f22fd91]
IDNF Pico       [6212d380693803381b1583a6]

HOW TO TEST

Add User ID in Line 41 in UserBlock to debug easier
![image](https://user-images.githubusercontent.com/37958655/154881515-4295458d-b367-4b31-9365-0195cb8fc62f.png)

Copy accessToken once login
![image](https://user-images.githubusercontent.com/37958655/154880767-32fddcfb-14b1-4cf5-ad4b-138a38a0949a.png)







Pasted inside Authorization-Bearer_Token



![image](https://user-images.githubusercontent.com/37958655/154880850-fd320d5d-d5a6-4fed-baf7-7d1a0c055794.png)


Try to follow and unfollow from [Following] list (same logic in [Followers] list, but [Following is easier to restore)
Once follow/unfollow, DO
GET http://localhost:5000/api/v1/following/follows/6204a777f82a1fb5c9f80a1f
To see the list of following match you expectation
![image](https://user-images.githubusercontent.com/37958655/154881175-b6d523fa-6bca-4b43-b917-a56b5c79906a.png)

If you accidently lost a relation, how to follow in postman
- From the same page without refreshing
- POST http://localhost:5000/api/v1/following/follow
- - {
    "actorId": "6204a777f82a1fb5c9f80a1f",
    "targetId": "{ID_YOU_WANT_TO_FOLLOW}"
}

Still fixing the E2E, but i wont affect the testing
